### PR TITLE
WIP - Migrate wiki page: IDE Settings

### PR DESCRIPTION
**Pull request by seanmadsen**
A simple migration of the IDE Settings wiki page
https://wiki.civicrm.org/confluence/display/CRMDOC/IDE+Settings+to+Meet+Coding+Standards

### CODE REVIEW

File: docs/setup/text-editor.md [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+3.  Open General > Editors > Text Editors.
+    1.  In Displayed tab width, enter 2.
+    2.  Click to enable Insert spaces for tabs.
+4.  Open General > Workspace.
+    1.  In Text file encoding, enable Other, and select UTF-8.
+    2.  In New text file line delimiter, enable Other, and select Unix.
+5.  Open PHP > Code Style > Code Templates.
+    1.  Download the attachment from this wiki page titled CiviCRM
+        codetemplates.xml
+    2.  Click Import, select CiviCRM codetemplates.xml from your local
```
Comment (totten) [inline docs/setup/text-editor.md]: This references an example file that doesn't seem to exist in the docs or the wiki page.

IMHO, remove the section. *Maybe* give a link to someone else's docs for Eclipse+Drupal (the same way as the other editors do).

**Comment by totten** [comment]
FWIW, `drupal.org` has a list of editor-specific directions under https://www.drupal.org/docs/develop/development-tools

I'm not sure we want to get into the game of maintaining docs for several IDEs. Here's what I'd rather communicate:

  1. Civi uses a variant of Drupal Coding Stanards. (Follow hyperlink to our docs of why's and what's.)
  2. The Civi coding standard is published as a formal ruleset based on [PHP Code Sniffer / `phpcs`](https://github.com/squizlabs/PHP_CodeSniffer).
  3. You can check code compliance on CLI using `civilint`.
  4. Provide an alphabetical/bulletted list of IDEs. For each one, give a link to instructions on configuring phpcs, e.g.
     a. [Atom](https://tommcfarlin.com/using-php-codesniffer/)
     b. [PHPStorm](https://www.jetbrains.com/help/phpstorm/2016.3/using-php-code-sniffer-tool.html)
     c. [Sublime Text](https://github.com/benmatselby/sublime-phpcs)

File: docs/setup/text-editor.md (hunk 1 above)
Comment (seanmadsen) [inline docs/setup/text-editor.md]: good catch. will fix.

**Comment by seanmadsen** [comment]
I like your suggested revisions, @totten. Will implement.

**Comment by totten** [comment]
Just want to make sure - per Mattermost discussion, you're working on another revision to this?

Main thought about currently posted draft: we're trying to refer them to external resources, but are those resources focused on "Drupal Coding Standard" (e.g. google for `phpstorm drupal coding standard`) or "phpcs" (e.g. google for `phpstorm phpcs`).

Orienting around `phpcs` should be more correct because it means that they can use *exactly* [the Civi ruleset](https://github.com/civicrm/coder/). The Drupal standard is very *close*, and in some IDE's the Drupal config may be perfectly compatible. But in some IDEs, the Drupal config would produce incorrect warnings because the Civi and Drupal standards are not exact matchs.

**Comment by seanmadsen** [comment]
Yeah @totten still work to-do on this. Thanks pointing that out re: Drupal standard vs phpcs.

**Comment by seanmadsen** [comment]
Ok after fiddling with this page a bit and thinking on it, I think it's best to just scrap it. @totten your feedback was good, and making the changes you suggested pretty much left me without much real content in this page, which pretty much agrees with the sentiment from @ErichBSchulz -- there's not much in here that should get migrated. 

The wiki has a page on PHP standards slated to bring over, and the content about `phpcs` will go in that page.

**Pull request closed without merging**